File: OdlEngine/CardTooltipGenerator/CardTooltip.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CardTooltipGenerator
{
    public class CardTooltip
    {
        public int CardId;
        public bool HasBlueprint;
        public List<string> Keywords;
        public List<int> RelatedCards;
    }
}

File: web-client/src/models/CardTooltip.js
export class CardTooltip {
  constructor(cardId, hasBlueprint = false, keywords = [], relatedCards = []) {
    this.cardId = cardId;
    this.hasBlueprint = hasBlueprint;
    this.keywords = keywords;
    this.relatedCards = relatedCards;
  }

  static fromJson(jsonData) {
    return new CardTooltip(
      jsonData.CardId,
      jsonData.HasBlueprint || false,
      jsonData.Keywords || [],
      jsonData.RelatedCards || []
    );
  }
} 